src/System.Security.Permissions/src/System/Security/CodeAccessPermission.cs

```diff
@@ -0,0 +1,20 @@
+namespace System.Security
+{
+    public abstract partial class CodeAccessPermission : System.Security.IPermission, System.Security.ISecurityEncodable, System.Security.IStackWalk
+    {
+        protected CodeAccessPermission() { }
+        public void Assert() { }
+        public abstract System.Security.IPermission Copy();
+        public void Demand() { }
+        public void Deny() { }
```

Comment:
In Mono, imperative CAS is not supported. They first check SecurityManager.SecurityEnabled, which is hard coded to always return FALSE. 
In Desktop, it always returns TRUE .. however, the Deny itself first checks AppDomain.CurrentDomain.IsLegacyCasPolicyEnabled and if that is false, throws NSE.

So in Mono it does nothing. In Desktop, it throws NSE unless you have CAS enabled. A Mono based app presumably would not, as it does not have IsLegacyCasPolicyEnabled and does not support imperative CAS.

To be compatible with Mono we should do nothing. Also note there is zero useage of the Deny()'s in our corpus.

---
In reply to: [70820609](https://github.com/dotnet/corefx/pull/10049#discussion_r70820609) [](ancestors = 70820609)